packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -176,31 +178,48 @@ class Hero extends StatefulWidget {
   /// left in place once the Hero shuttle has taken flight.
   final TransitionBuilder placeholderBuilder;
 
+  /// Whether to perform the hero transition if the [PageRoute] transition was
+  /// triggered by a user gesture such as a back swipe on iOS.
+  ///
+  /// If [Hero]s with the same [tag] on both the from and the to routes have
+  /// [transitionForUserGestures] set to true, a back swipe gesture will
```

Comment:
I agree. Though I'm not too sure what to put more specifically. 

I think didStartUserGesture is a bit of an implementation detail (in terms of how the HeroController receives events but we can envision changing that implementation detail without having to change the 'API' in the future).

Since I can't think of a specific other use case for the trigger and it's API stable now, it's probably better to be very specific in the doc and later expand than trying to restrict the API in the future.